config.go

```diff
@@ -0,0 +1,10 @@
+package plugin
+
+// Configor is the interface plugins should implement in order to provide configuration interface to the user
+type Configor interface {
+	Plugin
+	// DefaultConfig will be called on plugin initialization. Plugins should return a default configuration in YAML format. The default configuration will be provided to the user for future edit.
+	DefaultConfig() []byte
```

Comment:
How about we already provide functionallity for serialization and deserialization of the configuration. Each plugin must use yaml (otherwise it would be pretty difficult to make it configurable in the ui) so maybe something like this?`
```go
type Configor interface {
	Plugin
	// DefaultConfig will be called on plugin initialization. The default configuration will be provided to the user for future edit.
	DefaultConfig() interface{}
        // EmptyConfig returns an instance of an empty configuration.
        EmptyConfig() interface{}
	// LoadConfig will be called every time the plugin is initialized or the configuration has been changed by the user. 
        // Plugins should check whether the configuration is valid and optionally return an error. 
        // Parameter is guaranteed to be the same type as the return type of EmptyConfig()
	LoadConfig(interface{}) error
}
```

the plugin loader would then do something like this
```go
var plugin Configor
var confFromDB []byte

conf := plugin.EmptyConfig()
yaml.Unmarshal(confFromDB, conf)
plugin.LoadConfig(conf)
```

Reply:
Good idea.